helper/plan.php

```diff
@@ -453,57 +463,65 @@ public function previewHTML() {
      */
-    protected function stepThroughDocuments($type = self::TYPE_PAGES, &$skip = false) {
-        /** @var helper_plugin_move_op $MoveOperator */
-        $MoveOperator = plugin_load('helper', 'move_op');
+    protected function stepThroughDocuments($type = self::TYPE_PAGES, $skip = false) {
 
         if($type == self::TYPE_PAGES) {
             $file    = $this->files['pagelist'];
             $mark    = 'P';
             $call    = 'movePage';
-            $counter = 'pages_run';
+            $items_run_counter = 'pages_run';
         } else {
             $file    = $this->files['medialist'];
             $mark    = 'M';
             $call    = 'moveMedia';
-            $counter = 'media_run';
+            $items_run_counter = 'media_run';
         }
 
         $doclist = fopen($file, 'a+');
+        $log = "";
         for($i = 0; $i < helper_plugin_move_plan::OPS_PER_RUN; $i++) {
             $line = $this->getLastLine($doclist);
-            if($line === false) break;
+            if($line === false) {
+                break;
+            }
             list($src, $dst) = explode("\t", trim($line));
 
             // should this item be skipped?
-            if($skip) goto FINISH;
-
-            // move the page
-            if(!$MoveOperator->$call($src, $dst)) {
-                $this->log($mark, $src, $dst, false); // FAILURE!
-
-                // automatically skip this item if wanted...
-                if($this->options['autoskip']) goto FINISH;
-                // ...otherwise abort the operation
-                fclose($doclist);
-                return false;
+            if($skip === true) {
+                $skip = false;
             } else {
-                $this->log($mark, $src, $dst, true); // SUCCESS!
+            // move the page
+                if(!$this->MoveOperator->$call($src, $dst)) {
+                    $log .= $this->build_log_line($mark, $src, $dst, false); // FAILURE!
+
+                    // automatically skip this item only if wanted...
+                    if(!$this->options['autoskip']) {
+                        // ...otherwise abort the operation
+                        fclose($doclist);
+                        $return_items_run = false;
+                        break;
+                    }
+                } else {
+                    $log .= $this->build_log_line($mark, $src, $dst, true); // SUCCESS!
+                }
             }
 
             /*
              * This adjusts counters and truncates the document list correctly
              * It is used to finalize a successful or skipped move
              */
-            FINISH:
-            $skip = false;
+
             ftruncate($doclist, ftell($doclist));
-            $this->options[$counter]--;
-            $this->saveOptions();
+            $this->options[$items_run_counter]--;
+            $return_items_run = $this->options[$items_run_counter];
         }
+        $this->write_log($log);
+        $this->saveOptions();
```

Comment:
you moved saveOptions() outside the loop. When the move operation gets interrupted for whatever reason (timelimit hit, memory, etc...) it will pick up at the wrong place.